api/v1/verticadb_webhook.go

```diff
@@ -1261,6 +1285,57 @@ func checkInt64PtrChange(prefix *field.Path, fieldName string,
 	return allErrs
 }
 
+// checkImmutableSubclusterDuringUpgrade will ensure we don't scale, add or
+// remove subclusters during a replicated upgrade.
+func (v *VerticaDB) checkImmutableSubclusterDuringUpgrade(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
+	// This entire check can be skipped if we aren't doing replicated upgrade.
+	if oldObj.Status.UpgradeState == nil || len(oldObj.Status.UpgradeState.ReplicaGroups) == 0 {
+		return allErrs
+	}
+
+	// The subclusters defined in status.upgradeState.replicaGroups must stay
```

Comment:
I am not sure I understand this comment.

Reply:
The subclusters, whose names are stored in replicaGroups, must stay static throughout the upgrade. The size of the them cannot change. We cannot remove them. But, new subclusters may be added as this is something the upgrade process will do. So, we have to allow for that.